bot/exts/moderation/incidents.py

```diff
@@ -173,18 +265,33 @@ class Incidents(Cog):
         * If `Signal.ACTIONED` or `Signal.NOT_ACTIONED` were chosen, attempt to
           relay the incident message to #incidents-archive
         * If relay successful, delete original message
+        * Search the cache for the webhook message for this message, if found delete it.
         * See: `on_raw_reaction_add`
 
     Please refer to function docstrings for implementation details.
     """
 
+    # This dictionary maps an incident report message to the message link embed's ID
+    # RedisCache[discord.Message.id, discord.Message.id]
+    message_link_embeds_cache = RedisCache()
+
     def __init__(self, bot: Bot) -> None:
         """Prepare `event_lock` and schedule `crawl_task` on start-up."""
         self.bot = bot
 
+        self.bot.loop.create_task(self.get_webhook())
+
         self.event_lock = asyncio.Lock()
         self.crawl_task = self.bot.loop.create_task(self.crawl_incidents())
 
+    async def get_webhook(self) -> None:
+        """Fetch and store message link embeds webhook, present in #incidents channel."""
+        await self.bot.wait_until_guild_available()
+        self.incidents_webhook = await self.bot.fetch_webhook(Webhooks.incidents)
+
+        if not self.incidents_webhook:
+            log.error(f"Failed to fetch incidents webhook with id `{Webhooks.incidents}`.")
```

Comment:
This approach is problematic for a number of reasons.
* The attribute will not exist until the coro completes. This means that if a reaction triggers an event, it will fail on an attribute error. At the very least, code accessing this attribute must await the task.
* According to the docs, [`fetch_webhook`](https://discordpy.readthedocs.io/en/stable/api.html#discord.Client.fetch_webhook) will raise rather than return None if not found. This code assumes it will return None.
* If None is returned (in a hypothetical case, as per above), an error is logged, but code accessing the attribute will begin failing because it assumes the fetch to be successful.

Reply:
> This approach is problematic for a number of reasons.
> 
>     * The attribute will not exist until the coro completes. This means that if a reaction triggers an event, it will fail on an attribute error. At the very least, code accessing this attribute must await the task.
> 
>     * According to the docs, [`fetch_webhook`](https://discordpy.readthedocs.io/en/stable/api.html#discord.Client.fetch_webhook) will raise rather than return None if not found. This code assumes it will return None.
> 
>     * If None is returned (in a hypothetical case, as per above), an error is logged, but code accessing the attribute will begin failing because it assumes the fetch to be successful.

According to the second point, there is no need for doing this check, as it will raise when it fails to fetch the web-hook.

Reply:
I do not follow sorry. :smile:

Reply:
This doesn't solve the issues kwzrd outlined. What will happen is:
* An exception will be raised for the missing webhook.
* An exception will be raised for every single operation in the cog thereafter which tries to use this attribute.

I'm fine with an exception being logged in Sentry for the missing webhook, in which case the check above is not necessary, but the rest of the cog then should not assume the attribute exists, and only use the feature implemented in this PR if it has the value set (with a default of None set in the init).